torcharrow/velox_rt/numerical_column_cpu.py

```diff
@@ -606,18 +606,8 @@ def fill_null(self, fill_value: Union[dt.ScalarTypes, Dict]):
             raise TypeError(f"fill_null with {type(fill_value)} is not supported")
         if not self.is_nullable:
             return self
-        else:
-            # pyre-fixme[16]: Module `torcharrow` has no attribute `_torcharrow`.
-            col = velox.Column(get_velox_type(self.dtype))
-            for i in range(len(self)):
-                if self._getmask(i):
-                    if isinstance(fill_value, Dict):
-                        raise NotImplementedError()
-                    else:
-                        col.append(fill_value)
-                else:
-                    col.append(self._getdata(i))
-            return ColumnCpuMixin._from_velox(self.device, self.dtype, col, True)
+
+        return functional.coalesce(self, fill_value)
```

Comment:
nit: you need massage around the type nullability. Unfortunately `functional` now pessimistically makes returned column type always nullable, .e. g

```python
return functional.coalesce(self, fill_value)._with_null(False)
```

Perhaps also add a test on returned dtype in test? https://github.com/facebookresearch/torcharrow/blob/0a9eec6149659301cebb007cc409d4b307c382de/torcharrow/test/test_numerical_column.py#L420 ?

Reply:
Also remove the `_prototype_support_warning` on line 603 -- it's now use the native Velox C++ function.

Reply:
Actually , this doesn't work if you have a int32 column and try to fill_null with int (since Python int is interpreted as int64) . (caught by test) . Let's sync offline on this~